src/SpecFlow.NetCore/Args.cs

```diff
@@ -1,79 +1,91 @@
+ 		public const string ToolsVersionArgName = "--tools-version";
+ 
+ 		public string SpecFlowPath { get; }
+ 		public DirectoryInfo WorkingDirectory { get; }
+ 		public string TestFramework { get; }
+ 		public string ToolsVersion { get; }
+ 
+ 		public Args(string[] args)
+ 		{
+ 			WorkingDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
+ 
+ 			// Command line arguments are required.  If absent, exit.
+ 			if (args == null || !args.Any())
+ 			{
+ 				return;
+ 			}
+ 
+ 			// establish a dictionary of all good command line variables
+ 			var argDictionary = new Dictionary<string, string>()
 			{
-				if (IsOdd(i))
-					continue;
-
-				if (i + 1 >= args.Length)
-					throw new Exception("Uneven arguments");
-
-				var name = args[i];
-				var value = args[i + 1];
-
-				switch (name)
+ 				{SpecFlowPathArgName , null},
+ 				{WorkingDirectoryArgName, null },
+ 				{TestFrameworkArgName, null },
+ 				{ToolsVersionArgName, null }
+ 			};
+ 
+ 
+ 			string lastKey = null;
+ 			// loop through all arguments, attempting to detect and fix bad values
+ 			for (var x = 0; x<args.Length; x++)
+ 			{
+ 				if (argDictionary.ContainsKey(args[x]))
+ 				{
+ 					lastKey = args[x];
+ 				}
+				else if (args[x].Contains("--") || string.IsNullOrEmpty(lastKey))
 				{
-					case SpecFlowPathArgName:
-						SpecFlowPath = value;
-						break;
-
-					case WorkingDirectoryArgName:
-						if (!Directory.Exists(value))
-							throw new Exception("Working directory doesn't exist: " + value);
-						WorkingDirectory = new DirectoryInfo(value);
-						break;
-
-					case TestFrameworkArgName:
-						TestFramework = value;
-						break;
-
-					case ToolsVersionArgName:
-						ToolsVersion = value;
-						break;
-
-					default:
-						throw new Exception("Unknown argument: " + name);
+					// We are making the assumption that anything starting with -- is intentionally an argument key.
+					// If that argument key is not in the dictionary, we know it is a bad argument.
+					// Additionally, if the first argument key is not in our dictionary, the arguments are bad.
+					throw new Exception("Unknown argument: " + args[x]);
 				}
-			}
-
-			WriteLine("SpecFlowPath: " + SpecFlowPath);
-			WriteLine("WorkingDirectory: " + WorkingDirectory.FullName);
-			WriteLine("TestFramework: " + TestFramework);
-		}
-
-		private bool IsOdd(int i)
-		{
-			return i % 2 != 0;
-		}
+ 				else
+ 				{
+ 					argDictionary[lastKey] = string.IsNullOrEmpty(argDictionary[lastKey]) ? args[x] : argDictionary[lastKey] + args[x];
+ 				}
+ 			}
+ 
+ 			foreach(var key in argDictionary.Keys)
+ 			{
+ 				switch (key)
+ 				{
+ 					case SpecFlowPathArgName:
+ 						SpecFlowPath = argDictionary[key];
+ 						break;
+ 
+ 					case WorkingDirectoryArgName:
+ 						if (!Directory.Exists(argDictionary[key]))
+ 							throw new Exception("Working directory doesn't exist: " + argDictionary[key]);
+ 						WorkingDirectory = new DirectoryInfo(argDictionary[key]);

```

Comment:
This needs to be wrapped in a null check and only update the `WorkingDirectory` property if the arg has been supplied, otherwise it should continue to use the current directory as default. If you supply any other argument but not the `--working-directory`,  `argDictionary["--working-directory"]` remains `null` failing the `Directory.Exists()`.

![image](https://user-images.githubusercontent.com/2253814/35142753-23d9eff4-fd64-11e7-8dcd-aa8053909797.png)

Reply:
updated